torch/_inductor/codegen/cuda/cuda_kernel.py

```diff
@@ -504,7 +505,10 @@ def __init__(
         category: str,
         input_nodes: list[Buffer],
         layout: Layout,
-        make_kernel_render: Callable[[CUDATemplateBuffer, Optional[list[IRNode]]], str],
+        make_kernel_render: Callable[
+            [CUDATemplateBuffer, Optional[list[IRNode]]],
+            tuple[CUDATemplateKernel, functools.partial[str]],
```

Comment:
Is functool partial to restrictive here? Curious if the typing should be a Callable instead? Unless we are using specific functionality or hidden attrs only supported by partial functions.